homeassistant/components/automation/__init__.py

```diff
@@ -285,15 +285,21 @@ def is_on(self) -> bool:
 
         This method is a coroutine.
         """
-        if skip_condition or self._cond_func(variables):
-            self.async_set_context(context)
-            self.hass.bus.async_fire(EVENT_AUTOMATION_TRIGGERED, {
-                ATTR_NAME: self._name,
-                ATTR_ENTITY_ID: self.entity_id,
-            }, context=context)
-            await self._async_action(self.entity_id, variables, context)
-            self._last_triggered = utcnow()
-            await self.async_update_ha_state()
+        if not skip_condition and not self._cond_func(variables):
+            return
+
+        # Create a new context referring to the old context.
+        parent_id = None if context is None else context.id
+        trigger_context = Context(parent_id=parent_id)
+
+        self.async_set_context(trigger_context)
+        self.hass.bus.async_fire(EVENT_AUTOMATION_TRIGGERED, {
+            ATTR_NAME: self._name,
+            ATTR_ENTITY_ID: self.entity_id,
+        }, context=trigger_context)
```

Comment:
Shall use `context=context` here?

Reply:
No, we want the automation triggered to be with the new context, that way we will be able to attach automation metadata to the context.